src/main/java/com/microsoft/azure/functions/annotation/Retry.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.functions.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>
+ * Defines Retry Policy</p>
+ *
+ * @since 1.0.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Retry {
+    /**
+     * The strategy of retry
+     * @return The retry strategy
+     */
+    RetryStrategy strategy();
+    /**
+     * The maximum number of retries that will be attempted.
+     * @return The maximum retry count.
+     */
+    int maxRetryCount();
+    /**
+     * The minimum delay interval.
+     * @return The minimum retry delay.
+     */
+    String minimumInterval() default "";
```

Comment:
Shall we use integer here and set the interval in second directly? which may be more clear and no need to care about the format. Otherwise, do we support time string like `00:00:120`?

Reply:
I think the format is `00:00:10` . https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-error-pages?tabs=java

Reply:
thats a good point! minimumInterval, maximumInterval and delayInterval get converted to type [TimeSpan ](https://docs.microsoft.com/en-us/dotnet/api/system.timespan?view=netcore-3.1) , is there an equivalent in  Java?

Reply:
Duration?

Reply:
Not sure if converting `Duration` to string will give us the format `hh:mm:ss`

Reply:
We can do it. However, I'm not sure `Duration` is good fit for the annotation. Probably String (as is) or int (second) might be a good fit for it. (I just reply the TimeSpan equivalent) https://www.dariawan.com/tutorials/java/java-time-duration-tutorial-examples/
Hi @Flanker32 Which would you want to prefer? If you choose int, you need to format String. If you use String, no need to convert it since it is the format of `function.json`  IMO, String might be better. the reason is, Functions Host currently support String with that format. In the future, if people add expression on the String, we need to do breaking change. What do you think?